### DISCUSSION TITLE
Liabilities -looking for clarity

### DISCUSSION THREAD

**Discussion by skorvek** [firefly-iii/firefly-iii#6206] [Firefly III]
I've been having trouble setting up liability accounts, as mentioned in other posts I've made. I've read everything in the documentation and discussion I can find. Still confused as to what the "correct" thing to do here is.

I'll confine this question to one simple thing: a mortgage.  What is "best practice" in terms of Firefly in setting that up?

I've set it up like this:

- Liability account for the mortgage configured as a mortgage and I owe this debt
- Expense account for mortgage interest
- Asset account (checking) where the funds come for the payment
- Categories, tags, bill, budget also set up (only tangentially relevant to this discussion)

I make a withdrawal as per the documentation- source asset account destination liability for the mortgage payment.  My asset account balance goes down, it's recorded as a withdrawal, the transaction is red with a negative sign in the asset account.

On the liability side- the transaction line shows as a withdrawal (arrow pointing to the left) with a red amount negative signed.  That's really confusing.  But the graph line slopes up to the right, liability balance is moving toward zero.

But the balance is now off- that transaction didn't go all toward principal, some went toward interest. so I have two options:

- Split the transaction - principal to loan and interest toward interest.
- Enter an interest-only transaction for the interest charge: source is liability, destination is expense account.  

Which should I choose?  

If I do the split- I cannot automate it (cannot use the rules to split the imported transaction), ~~and I cannot assign the principal portion to a budget or a bill~~ (I can as long as the splits are withdrawals).  So those views would be inaccurate. 

If I do an interest only transaction to account for the additional liability, then the liability page is off.  That's also red negative signed, withdrawal, and the monthly "spent" amounts are also incorrect. This is the only way I can use a recurring transaction though- I did try a recurring transaction with a dummy amount for the interest (0.01) so I know to go back to update it.

**Comment by JC5** [comment]
I do the first thing myself, a split transaction.

**Comment by skorvek** [comment]
I made a few clarifications/edits in my post above to correct some misstatements- I was thinking of when I tried transfers following the asset/credit card model (and the fixed payment issue in another post).

As I continue to test, it seems like the best way for fixed liabilities (mortgages, car payments) here is to use a transaction with the split withdrawals, so that only the principal portion goes to the liability, completely leaving the interest portion out of the liability account. That gets diverted to the expense account by one of the splits.

Then I am using two withdrawals- so I can tie them together with the budget and the liability balance is correct.

However, that is REALLY HARD to do. Turns out out cannot create a split recurring transaction, and as noted by others, we cannot import splits via the importer (in my case, doesn't matter as the transaction isn't split in the import file).  And since you cannot split a transaction using rules, this is 100% hand entry every month. I have to clone an existing transaction (it's way too much metadata to ensure I don't forget a bit of it), edit the amounts, and then save, and delete the original transfer.  Then do that again each transaction for each liability. I pay my mortgage biweekly and have a few properties and other liabilities, so that really adds up to a lot of hand work.

Still working on how to easily report an overview of the entire liability- total cost, % interest, LTV, etc. I'll post more when I get that figured out.

I will note this is not following the documentation of posting a withdrawal to the liability to account for interest, or "paying the loan shark" which is what ends up with odd negative amounts in the liability.

**Reply by JC5** [reply]
I'm aware of the issue and working on a solution.

**Comment by marcelbrueckner** [comment]
I'm wrapping my head around liabilities as well as I want to enter my loan into Firefly III in order to keep track of the remaining amount. That's how I wanted to handle it.

1. I've created a liability, starting at 0.
2. Whenever I have to pay a bill in the context of what I've taken the loan for, I create an expense from my liability to my asset account (I loan money from my bank). From my asset account, I pay the bill. This repeats as long as there are bills to pay. This might take some months until all bills are due.
3. In the meantime, I have to pay interest for the money I've borrowed, thus creating an expense from my loan to my Bank's expense account (and will further do so every month).
4. Each month, I will transfer money from my asset account (checking account) to the liability. This covers the respective month's interest and also pays down some of the liability. For this, I will need to split my transaction. This repeats until the loan has been paid off.

Now my issue with this is the following:

The interest transaction I create in (3) I categorize as interest payment. When moving money from my asset account to the liability (4), this is also an expense which I could/should categorize. This, however, would result in two transactions categorized as interest, although it's only paid once in this case. The transaction from asset to liability should be a transfer instead.

So I came to the conclusion to either do as @skorvek proposes:

> it seems like the best way for fixed liabilities (mortgages, car payments) here is to use a transaction with the split withdrawals, so that only the principal portion goes to the liability, completely leaving the interest portion out of the liability account. That gets diverted to the expense account by one of the splits.

Or dismiss liabilities at all and track them with an extra asset account instead. This way, transactions that move money from my checking account to my "loan account" are handled as transfers. Transactions from the "loan account" to my bank's expense account can be categorized accordingly.

So far a mix of both would be best?

**Reply by JC5** [reply]
> it seems like the best way for fixed liabilities (mortgages, car payments) here is to use a transaction with the split withdrawals, so that only the principal portion goes to the liability, completely leaving the interest portion out of the liability account. That gets diverted to the expense account by one of the splits.

This is the best way atm. I'm not entirely happy yet with the setup but I haven't had the time to think about simplifications.
